### PR TITLE
Spell

### PR DESCRIPTION
**Pull request by calcitem**
* .gitattributes: Do not always perform LF normalization on .txt and .json files

* ci: Add spell-check-this
  - Copy https://github.com/check-spelling/spell-check-this/tree/main/.github
  - Update spelling/excludes.txt
  - Copy words of .vscode/settings.json -> "cSpell.words" to expect list
  - Only check C++ and Dart files
  - Exclude feedback_localization.dart
  - Do not check Qt words
  - Fill allow list with C++ 17 & Dart & Qt keywords
 
* Fix some spelling errors

### CODE REVIEW

**Comment by calcitem** [comment]
LGTM, Rebase, do not squash.

File: src/ui/flutter_app/lib/services/storage/adapters/color_adapter.dart [hunk 1]
```diff
@@ -23,7 +23,7 @@ import 'package:json_annotation/json_annotation.dart' show JsonSerializable;
 /// Color Adapter
 ///
 /// This adapter provides helper functions to be used with [JsonSerializable]
-/// and is a general [TypeAdapter] to be used with Hive [Box]ses
+/// and is a general [TypeAdapter] to be used with Hive [Box]es
```
Comment (calcitem) [inline src/ui/flutter_app/lib/services/storage/adapters/color_adapter.dart:26]: Please review this line.

Comment (Leptopoda) [reply src/ui/flutter_app/lib/services/storage/adapters/color_adapter.dart:26]: this is just a doc comment so it won't affect the app at all

**Comment by calcitem** [comment]
These commits' modification points are relatively scattered, it is recommended to rebase as soon as possible. Otherwise, there may be more and more conflicts.

**Comment by calcitem** [comment]
I found a method to reduce typo:

https://www.howtogeek.com/429702/how-to-enable-text-prediction-for-a-hardware-keyboard-on-windows-10/

**Pull request merged**